.github/workflows/test.yml

```diff
@@ -0,0 +1,145 @@
+name: codechecker-tests
+
+# Triggers the workflow on push or pull request events.
+on: [push, pull_request]
+
+jobs:
+  lint:
+    name: Linters (pylint, pycodestyle)
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Install dependencies
+        run: |
+          pip install $(grep -iE "pylint|pycodestyle" analyzer/requirements_py/dev/requirements.txt)
+      - name: Run tests
+        run: make pylint pycodestyle
+
+  analyzer:
+    name: Analyzer
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+
+      - name: Install dependencies
+        run: sh .github/workflows/install-deps.sh
+
+      - name: Build the package
+        run: |
+          make pip_dev_deps
+          BUILD_UI_DIST=NO make package
+
+      - name: Run analyzer tests
+        run: |
+          make -C analyzer \
+            test_unit \
+            test_functional \
+            test_tu_collector \
+            test_merge_clang_extdef_mappings \
+            test_statistics_collector \
+            test_build_logger
+
+  web:
+    name: Web
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    strategy:
+      matrix:
+        database: [sqlite, psql_pg8000, psql_psycopg2]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+
+      - name: Install dependencies
+        run: sh .github/workflows/install-deps.sh
+
+      - name: Init .pgpass
+        run: |
+          echo '*:*:*:*:postgres' > $HOME/.pgpass
+          chmod 0600 $HOME/.pgpass
+
+      - name: Run tests
+        env:
+          PGPASSWORD: postgres
+        run: |
+          export PGPASSFILE=$HOME/.pgpass
+
+          make pip_dev_deps
+          BUILD_UI_DIST=NO make package
+
+          make -C tools/plist_to_html test
```

Comment:
Maybe the tools (plist to html, report_converters, tu_collector ...) tests could be executed as a separate job?